analyzers/src/SonarAnalyzer.Common/Rules/Utilities/TokenTypeAnalyzerBase.cs

```diff
@@ -131,15 +134,15 @@ token switch
                     _ => null,
                 };
 
-            protected static TokenInfo TokenInfo(SyntaxToken token, TokenType tokenType) =>
-                tokenType == TokenType.UnknownTokentype
-                || (string.IsNullOrWhiteSpace(token.Text) && tokenType != TokenType.StringLiteral)
-                    ? null
-                    : new()
-                    {
-                        TokenType = tokenType,
-                        TextRange = GetTextRange(token.GetLocation().GetLineSpan()),
-                    };
+            protected TokenInfo TokenInfo(SyntaxToken token, TokenType tokenType)
+            {
+                var span = token.GetLocation().GetMappedLineSpanIfAvailable();
+                return tokenType == TokenType.UnknownTokentype
+                    || (string.IsNullOrWhiteSpace(token.Text) && tokenType != TokenType.StringLiteral)
+                    || !(string.IsNullOrWhiteSpace(filePath) || string.Equals(span.Path, filePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
Shouldn't return null if the `filePath` is null or whitespace?

Reply:
I made that choice for two reasons:
1. convenience: [ClassifierTestHarness.AssertTokenTypes](https://github.com/SonarSource/sonar-dotnet/pull/7924/files#diff-3f5ca57ceaf973ce63f2dc5c5680bf6d993851d546425cab194ecb988f4027b0R33) instantiates a `TokenClassifier` instance and a `TriviaClassifier` instance without having at hand the file path, which is implicitly created by [ParseTokens](https://github.com/SonarSource/sonar-dotnet/pull/7924/files#diff-3f5ca57ceaf973ce63f2dc5c5680bf6d993851d546425cab194ecb988f4027b0R30)
2. safety: while it seems indeed reasonable, to always return `null` in absence of `filePath`, I didn't know whether a `SyntaxTree` may have, for any reason, no file path (as it was somehow fully compiled in-memory), and I didn't want to alter the behavior the `TokenTypeAnalyzer` had before these changes

If we can/want to assume that the `TokenTypeAnalyzer` should always require `filePath`, we should do some refactoring to the `ParseTokens`, to basically return `snippet0.cs`, which would then be passed to the classifier.